### DISCUSSION TITLE
Can i make only htmx web page without using any javascript?

### DISCUSSION THREAD

**Discussion by tusuii** [bigskysoftware/htmx#1552] [Polls]
i am thinking to test it out and try out myself on my projects

**Comment by Telroshan** [comment]
It really is one of those "it depends" situations imo. What it depends on exactly, is what project you're working on.

Htmx does all the work for communicating with the server and swapping content.
For a simple website, you may not need any additional handwritten JavaScript on top of htmx.
However, htmx being a "server-oriented" lib, it is not designed neither capable for pure client side work for interactivity and UX.

For the latter aspect though, you may want to check out [_hyperscript](https://hyperscript.org/) which was designed by the same person, and this one is for client-side work.
It's a common combo to use both htmx & hyperscript. You wouldn't write "any javascript" in that you would be writing hyperscript syntax in an html attribute, not JS so to speak

For a drag'n drop behaviour for example, you would need a bit of handwrittenJS, or hyperscript.

Though, is it problematic? I'd argue not.
In my opinion, writing javascript is perfectly sane & fine when it's done for pure client-side purposes, such as interactivity or user experience. What sucks is when you have to write JS to communicate with your backend, and for that, htmx perfectly does its job, you won't need any handwritten JS for such things.

So to answer your question: **yes**, you can make web pages with htmx only and without using any javascript, but you might want to add interactivity/UX then using JS, hyperscript or anything else, which shouldn't be a bother though as those were designed for that.
